actions/views/browser.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {GenericAction} from 'mattermost-redux/types/actions';
+
+import {Constants, ActionTypes, WindowSizes} from 'utils/constants';
+
+export function emitBrowserWindowResized(): GenericAction {
+    const width = window.innerWidth;
+
+    let windowSize;
+    switch (true) {
+    case width > Constants.DESKTOP_SCREEN_WIDTH: {
+        windowSize = WindowSizes.DESKTOP_VIEW;
+        break;
+    }
+    case width > Constants.TABLET_SCREEN_WIDTH && width <= Constants.DESKTOP_SCREEN_WIDTH: {
+        windowSize = WindowSizes.SMALL_DESKTOP_VIEW;
+        break;
+    }
+    case width > Constants.MOBILE_SCREEN_WIDTH && width <= Constants.TABLET_SCREEN_WIDTH: {
+        windowSize = WindowSizes.TABLET_VIEW;
+        break;
+    }
+    case width <= Constants.MOBILE_SCREEN_WIDTH: {
+        windowSize = WindowSizes.MOBILE_VIEW;
+        break;
+    }
+    }
+
+    return {
+        type: ActionTypes.BROWSER_WINDOW_RESIZED,
+        data: windowSize || WindowSizes.DESKTOP_VIEW,
+    };
```

Comment:
instead of returning a value and alternative you could just use the default case for that, so that the variable is always "filled"

Reply:
TL;DR, I'll go with switch's default using fall through.

It's a little frustrating when it comes to conditional logic in JS. Everyone seems to have a different preference/opinion for how to do it "right" :P

In this case you could:
1. Just set the default when defining the variable and be done with it.
2. Use switch's default with fall through
3. Use switch's default without fall through
4. Set the default value at the end if none has been set already

I actually prefer to use switch's default option with fall through in a situation like this, but I've seen many comments where people prefer not to take that approach because of the fall through ... even though fall through is a very viable approach and can be super helpful in the right situation. You could omit the case, and thereby remove fall through altogether given it's the default case, but I prefer to explicitly identify all the cases in something like this so there is no question later on down the road as to what the original intentions were. It's much easier to see what the case is than to have to try to infer it by understanding what all the other cases do first.

Reply:
Well, I guess fall through is explicitly frowned upon in our linting. :P